projects/sweep/index.js

```diff
@@ -0,0 +1,43 @@
+const sdk = require('@defillama/sdk')
+const ADDRESSES = require('../helper/coreAssets.json');
+const SWEEP_ADDRESS = '0xB88a5Ac00917a02d82c7cd6CEBd73E2852d43574';
+
+const config = {
+  ethereum: {
+    fromBlock: 18017036,
+  },
+  arbitrum: {
+    fromBlock: 124179091,
+  },
+  optimism: {
+    fromBlock: 110684392,
+  }
+};
+
+async function tvl(_a, _b, _c, { api }) {
+  const balances = {}
+  const USDC_ADDRESS = ADDRESSES[api.chain].USDC;
+  const minters = await api.multiCall({ abi: 'address[]:getMinters', calls: [SWEEP_ADDRESS] });
+
+  const balanceOfCalls = minters[0].map((minter) => {
+    return api.multiCall({
+      abi: 'function balanceOf(address) external view returns(uint256)',
+      calls: [{ target: USDC_ADDRESS, params: minter }]
+    });
+  });
+
+  const assetValueCalls = minters[0].map((minter) => {
+    return api.multiCall({ abi: 'uint256:assetValue', calls: [minter] });
+  });
+
+  const valueCalls = await Promise.all([...balanceOfCalls, ...assetValueCalls]);
```

Comment:
wait, wouldnt the `assetValueCalls` and `balanceOfCalls` have duplicated value? like same USDC is counted under both?

Reply:
assetValueCalls only looks at holdings that are not USDC nor SWEEP. It represents the value of an investment in an external protocol, like USD-Plus tokens or sDAI tokens.
Code example here: https://github.com/SweeprFi/sweepr-contracts/blob/main/contracts/Assets/USDPlusAsset.sol#L62
Deployment of that here: https://arbiscan.io/address/0x52D0a9E74fC159F47cEE668801082a975c10bBba